primitives/src/runtime_api.rs

```diff
@@ -135,7 +135,12 @@ sp_api::decl_runtime_apis! {
 
 		/// Yields information on all availability cores as relevant to the child block.
 		/// Cores are either free or occupied. Free cores can have paras assigned to them.
-		fn availability_cores() -> Vec<CoreState<H, N>>;
+		fn availability_cores() -> Vec<crate::v4::CoreState<H, N>>;
+
+		/// Yields information on all availability cores as relevant to the child block.
+		/// Cores are either free or occupied. Free cores can have paras assigned to them.
+		#[api_version(5)]
+		fn availability_cores_on_demand() -> Vec<vstaging::CoreState<H, N>>;
```

Comment:
So this only gets us on-demand availability cores? I guess not, from the docs and from my understanding. So where does the name come from?

Reply:
It's supposed to be the `v5` `availability_cores` call. I thought just calling it `availability_cores_v5` would be a bit meaningless but I see how the current name is misleading as well.